src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -353,18 +351,6 @@ await foreach (ConfigurationSetting setting in _client.GetConfigurationSettingsA
                                             ((exception as AggregateException)?.InnerExceptions?.All(e => e is RequestFailedException) ?? false) ||
                                             exception is OperationCanceledException)
             {
-                if (_options.OfflineCache != null)
-                {
-                    // During startup or refreshAll scenario, we'll try to populate config from offline cache, if available
-                    cachedData = _options.OfflineCache.Import(_options);
-                    
-                    if (cachedData != null)
-                    {
-                        data = JsonSerializer.Deserialize<IDictionary<string, ConfigurationSetting>>(cachedData);
-                    }
-                }
-
-                // If we're unable to load data from offline cache, check if we need to ignore or rethrow the exception 

```

Comment:
Looks like `data` will always be null now.

Reply:
So could add `ignoreFailures` to the when block.

Reply:
We still need to catch the specific exceptions regardless of `ignoreFailures`. If `ignoreFailures` is added to the when block filtering, we will end up throwing the exceptions to the caller since no matching catch block is present to handle those exceptions.